### DISCUSSION TITLE
No outline? Shows "No symbols found in document" for c ?

### DISCUSSION THREAD

**Discussion by NorskieRider** [codefori/vscode-ibmi#1103] [IBM i]
When I browse .c files through the IBM i objects view the outline shows "No symbols found in document'. The language mode does show that it is 'c'. 

This works properly through the Explorer, so it appears to be unique to the IBM i object browser. 

Does code for IBM i have it's own outline symbol parser or does it use the standard vscode one? Is there a way to configure which one it uses?

**Comment by worksofliam** [comment]
We don't add any special language tools for C/CPP. 

Are you saying that when you open a member with .c as the extension, the language isn't detected? (as it would show C where the language id is show in the bottom right)

**Reply by NorskieRider** [reply]
When I open a member with .c as the extension it detects the language id as 'C' in the bottom right. But the outline pane shows "No symbols found". 

Same with .cmd, .rpg and .pf but I don't care as much about those.

When I open members with .rpgle, .clp or .clle as the extension it detects those language ids correctly, and an outline is displayed.

**Reply by worksofliam** [reply]
Right. We don't provide any C language tools because you can find them on the extension marketplace already.

For C:

* just the basics: anylang with C
* more complex: C\C++ from Microsoft

**Reply by NorskieRider** [reply]
I have them installed. It works correctly when viewing .c files outside of Code for IBM i ... i.e. through Explorer on local files. 

How do I configure Code for IBM i to use them with the members?

**Reply by worksofliam** [reply]
Huh, I've never had an issue there. It's usually tied to the language id.

Is there any output views with errors related to the language providers you're using?

Perhaps they don't support the member file system correctly.

**Reply by NorskieRider** [reply]
I appreciate the help!

Hm, I noticed that when I open a .c in explorer, next to the language id it shows 'Win32' as the "C/C++ Configuration". When I open a .c member that position on the status line goes away (i.e. it doesn't show Win32 or anything else). 

Regarding the output, I see this in the 'Code for IBM i' but it doens't look like any error:

```
.: system "CPYTOSTMF FROMMBR('/QSYS.lib/FSTESTS.lib/QCSRC.file/UPDCSEDTA2.mbr') TOSTMF('/tmp/vscodetemp-O_XsNoaeCD') STMFOPT(*REPLACE) STMFCCSID(1208) DBFCCSID(*FILE)"
{
    "code": 0,
    "signal": null,
    "stdout": "CPCA082: Object copied.",
    "stderr": ""
}
```
I didn't see anything that was remotely associated with this, errors or otherwise.

When you write "they don't support" who is "they" ? vscode or Code for IBM i ?

**Reply by NorskieRider** [reply]
I'm still struggling with this. There are no outlines for C or CPP parts. 

This may be a VSCode configuration issue rather than Code for IBM i issue. As I'm new to VSCode I don't know where to look or how to resolve this, so any assistance would be appreciated.

**Reply by chrjorgensen** [reply]
@NorskieRider @worksofliam FWIW, I can recreate this issue - opening a C streamfile does not provide any content in the outline, but downloading it locally and opening it from there, the outline is showing symbols etc.

So it's both members and streamfiles, that can't be read by the C language extensions. I tried the following two extensions:
[Microsoft C/C++ for Visual Studio Code](https://marketplace.visualstudio.com/items?itemName=ms-vscode.cpptools)
[clangd](https://marketplace.visualstudio.com/items?itemName=llvm-vs-code-extensions.vscode-clangd)

@NorskieRider Best advice is to develop locally until the issue has been resolved, if possible...

**Reply by worksofliam** [reply]
I've had luck with [anycode](https://marketplace.visualstudio.com/items?itemName=ms-vscode.anycode), which I believe does work for members and streamfiles, but it really only looks at the active source.

**Reply by NorskieRider** [reply]
anycode worked! The other onces (clangd and MS C/C++) .. and various others ... did not.

**Reply by NorskieRider** [reply]
Came back to vscode after some time in other IDE's and anycode no longer worked, so unchecking that.
